src/server/wesnothd/server.hpp

```diff
@@ -77,10 +77,13 @@ class server : public server_base
 	}
 	void send_to_lobby(simple_wml::document& data, std::optional<player_iterator> exclude = {});
 	void send_to_player(player_iterator player, simple_wml::document& data) {
-		utils::visit(
-			[this, &data](auto&& socket) { async_send_doc_queued(socket, data); },
-			player->socket()
-		);
+		if(player_connections_.get<socket_t>().find(player->socket()) != player_connections_.end())
```

Comment:
You cannot use the `player_iterator` since the issues is that the iterator becomes invalid. `fuh::async_get_and_send_game_history` needs to be adjusted to not keep the iterator but just the socket and then use the socket to send. You can either put the code in that function or add another send_to_player function that takes a socket and checks it against `player_connections_` like you do here. The existing send_to_player function should not be changed.

Reply:
Updated.